pkg/wallet/storage.go

```diff
@@ -0,0 +1,139 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+//nolint
+package wallet
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/bluele/gcache"
+
+	"github.com/hyperledger/aries-framework-go/component/storage/edv"
+	"github.com/hyperledger/aries-framework-go/pkg/crypto"
+	"github.com/hyperledger/aries-framework-go/pkg/crypto/tinkcrypto"
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/packer"
+	"github.com/hyperledger/aries-framework-go/pkg/doc/jose"
+	"github.com/hyperledger/aries-framework-go/pkg/kms"
+	"github.com/hyperledger/aries-framework-go/spi/storage"
+)
+
+// storageProvider all wallet content store storage provider operations.
+// if profile has EDV settings then call to OpenStore will return EDV store instance by creating new EDV provider.
+// Otherwise default provider will be used to open store.
+// (Refer #2745 for more details)
+type storageProvider struct {
+	profile         *profile
+	defaultProvider storage.Provider
+}
+
+func newWalletStorageProvider(profile *profile, provider storage.Provider) *storageProvider {
+	return &storageProvider{profile: profile, defaultProvider: provider}
+}
+
+// OpenStore opens and returns store and sets store config to provider.
+// if wallet profile has EDV settings then auth provided will be used to initialize edv storage provider.
+func (s *storageProvider) OpenStore(auth string, config storage.StoreConfiguration) (storage.Store, error) {
+	var provider storage.Provider
+	var err error
+
+	if s.profile.EDVConf != nil {
+		provider, err = createEDVStorageProvider(auth, s.profile.EDVConf)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		provider = s.defaultProvider
+	}
+
+	store, err := provider.OpenStore(s.profile.ID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open store : %w", err)
+	}
+
+	err = provider.SetStoreConfig(s.profile.ID, config)
+	if err != nil {
+		e := store.Close()
+		if e != nil {
+			logger.Warnf("failed to close store: %s", e)
```

Comment:
Any reason why you don't return this error?

Reply:
otherwise it will eatup original error of setting store config.